### ISSUE TITLE
newline-before-return false positive

### ISSUE THREAD

**Issue by YKlos** [palantir/tslint#2311]
### Bug Report

- __TSLint version__: 4.5.1
- __TypeScript version__: 2.2.1
- __Running TSLint via__: Node.js API

#### TypeScript code being linted

```ts
class HelloWorld {

    test() {
        console.log("Any statement");
        // Any comment

        return;
    }
}
```

with `tslint.json` configuration:

```json
"rules":
{
    "newline-before-return": true
}
```

#### Actual behavior
If there is a comment between the last statement and the return statement, tslint throws a false positive "Missing blank line before return"

**Comment by sbj42** [comment]
Here's another case not covered by the rule:
```ts
function foo() {
    fn();
    // comment

    // comment
    return;
}
```

The `if (comments) { ... }` block in `newlineBeforeReturnRule.ts` should probably check if:
- There's a blank line in between one comment and another
- Or there's a blank line after the last comment

Either of those conditions could trigger an early return and no warning.
